### PR TITLE
Autolog Keras summaries as an artifact

### PR DESCRIPTION
**Pull request by juntai-zheng**
In this PR, Keras model summaries are autologged as artifacts (in addition to it currently being logged as a tag). Also, `summary` is changed to `model_summary` for both tags and artifacts.

image

## How is this patch tested?

Tests were updated to check for the summary existing as an artifact.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

With `autolog()`, Keras model summaries are now also logged as artifacts (in addition to being logged as tags).

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [X] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/keras.py [hunk 1]
```diff
@@ -400,6 +400,11 @@ def on_train_begin(self, logs=None):  # pylint: disable=unused-argument
             summary = '\n'.join(sum_list)
             try_mlflow_log(mlflow.set_tag, 'summary', summary)
 
+            with open('summary.txt', 'w') as f:
```
Comment (smurching) [inline mlflow/keras.py]: Maybe write to a temporary file created via [`tempfile.mkstemp()`](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp), just to be safe/guarantee that we don't overwrite files in the user's working directory

**Review by smurching — COMMENTED** [review]
One comment otherwise LGTM!

File: mlflow/keras.py (hunk 1 above)
Comment (juntai-zheng) [reply mlflow/keras.py]: Good catch, will do

**Review by smurching — APPROVED** [review]
LGTM

**Pull request merged**